### ISSUE TITLE
[JSON API] Incorrect response format when requesting resources

### ISSUE THREAD

**Issue by nickschot** [fortunejs/fortune#91]
I'm doing a request like `/books/:ID/pages`. When the result is a single page, fortune sends back a response with an Object `data`:
```
{  
    "links":{  
        "self":"/pages/d6ede91990dd1f9bf64a068684920fe5"
    },
    "data":{  
        "type":"pages",
        "id":"d6ede91990dd1f9bf64a068684920fe5",
        "attributes":{  
            "name": "This is a page"
        },
        "links":{  
            "self":"/questions/d6ede91990dd1f9bf64a068684920fe5"
        }
    }
}
```

I'm not sure whether it's ember-data or fortune at fault here, because ember-data expects `data` to be an array of pages with this kind of request.

JSON API states: A server MUST respond to a successful request to fetch a resource collection with an array of resource objects or an empty array ([]) as the response document's primary data.

I'm guessing /books/:ID/pages is a request to a collection which means the result should be an array even though the result is a single object.

JSFiddle for demonstration: http://jsfiddle.net/27v46pvu/1/ When you make an Array of the data object in the mockjax at the bottom, it starts working correctly. Else (with canary ember-data) it throws an error about record.map. With ember-data 1.13.5 it silently fails and never returns any `pages`.

See also: https://github.com/emberjs/data/issues/3556 (my simplified example there is most likely wrong as there is an array there).

**Comment by nickschot** [comment]
Updated the link to the jsfiddle, it was wrong.

**Comment by 0x8890** [comment]
This seems like a bug in the serializer. Could you clarify if `data` needs to be an array even if the relationship is singular?

**Comment by nickschot** [comment]
A server MUST respond to a successful request to fetch an individual resource with a resource object or null provided as the response document's primary data.

It's under the the fetching resources section. I'd say the relationship should be checked to see which one is needed (-to-one or -to-many)

**Comment by 0x8890** [comment]
Fixed in commit ba8a43727a5a52de1a86c8eaedf8d504a56dd39b.
